transport-native-epoll/src/main/java/io/netty/channel/epoll/AIOContext.java

```diff
@@ -31,12 +33,18 @@
 
     private final long address;
     private long nextId;
+    private long outstanding;
+    private long maxOutstanding;
     private final Map<Long, IORequest> outstandingRequests;
+    private final ArrayDeque<IORequest> pendingRequests;
```

Comment:
We are definitely OK growing this indefinitely?

Reply:
It's limited to 64k entries below

Reply:
I don't think so. From the doc: `Array deques have no capacity restrictions`; as far as I understand, the parameter in the constructor is just the initial value.